src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/SpanDataExtensions.cs

```diff
@@ -0,0 +1,264 @@
+                otlpResources.Attributes.AddRange(
+                    resource.Key.Attributes.Select(ToOtlpAttribute));
+
+                var otlpResourceSpans = new OtlpTrace.ResourceSpans
+                {
+                    Resource = otlpResources,
+                };
+                otlpResourceSpans.InstrumentationLibrarySpans.AddRange(libraryList);
+
+                resourceSpansList.Add(otlpResourceSpans);
+            }
+
+            return resourceSpansList;
+        }
+
+        internal static OtlpTrace.Span ToOtlpSpan(this SpanData spanData)
+        {
+            // protobuf doesn't understand Span<T> yet: https://github.com/protocolbuffers/protobuf/issues/3431
+            Span<byte> traceIdBytes = stackalloc byte[16];
+            Span<byte> spanIdBytes = stackalloc byte[8];
+
+            spanData.Context.TraceId.CopyTo(traceIdBytes);
+            spanData.Context.SpanId.CopyTo(spanIdBytes);
+
+            var parentSpanIdString = ByteString.Empty;
+            if (spanData.ParentSpanId != default)
+            {
+                Span<byte> parentSpanIdBytes = stackalloc byte[8];
+                spanData.ParentSpanId.CopyTo(parentSpanIdBytes);
+                parentSpanIdString = ByteString.CopyFrom(parentSpanIdBytes.ToArray());
+            }
+
+            var otlpSpan = new OtlpTrace.Span
+            {
+                Name = spanData.Name,
+
+                Kind = spanData.Kind == null
+                    ? OtlpTrace.Span.Types.SpanKind.Unspecified
+                    : (OtlpTrace.Span.Types.SpanKind)spanData.Kind.Value,
+
+                TraceId = ByteString.CopyFrom(traceIdBytes.ToArray()),
+                SpanId = ByteString.CopyFrom(spanIdBytes.ToArray()),
+                ParentSpanId = parentSpanIdString,
+
+                Status = ToOtlpStatus(spanData.Status),
+
+                StartTimeUnixNano = spanData.StartTimestamp.ToUnixTimeNanoseconds(),
+                EndTimeUnixNano = spanData.EndTimestamp.ToUnixTimeNanoseconds(),
+            };
+
+            if (!ReferenceEquals(spanData.Attributes, EmptySpanDataAttributes))
+            {
+                otlpSpan.Attributes.AddRange(spanData.Attributes.Select(ToOtlpAttribute));
+
+                // TODO: get dropped count.
+            }
+
+            if (!ReferenceEquals(spanData.Events, EmptySpanDataEvents))
+            {
+                otlpSpan.Events.AddRange(spanData.Events.Select(ToOtlpEvent));
+
+                // TODO: get dropped count.
+            }
+
+            if (!ReferenceEquals(spanData.Links, EmptySpanDataLinks))
+            {
+                otlpSpan.Links.AddRange(spanData.Links.Select(ToOtlpLink));
+
+                // TODO: get dropped count.
+            }
+
+            return otlpSpan;
+        }
+
+        private static OtlpTrace.Status ToOtlpStatus(Status status)
+        {
+            // At this stage Status.IsValid is always true, just add status message if !Ok
+            if (status == Status.Ok)
```

Comment:
OK may have a description. Why not add it?

Reply:
This is comparing the whole struct so if a description was set it will be included. I will add a test to ensure that the behavior is not broken in the future.

Reply:
Actually, I'm not finding a way to set the description. Opened issue #653.